### PR TITLE
task-12 validation

### PR DESCRIPTION
**Pull request by DiscoJordan**
Task-12: Add validation

It's good if we always receive a request on our server in the same form and we know how to process it, but what if an empty or incomplete request comes to us, and we cannot make a request to a third-party source and form a decent response to the user, or due to the incompleteness of the request, our business logic breaks down.

To do this, we can use validation to exclude such cases, and then meaningful errors will come to the client's side, indicating what we need in order to fulfill their request.

We choose to use such a wonderful library as Joi.

Acceptance criteria:

Connect the library
Learn what is Middleware
Write a generic function to validate incoming parameters on requests
Add validation to existing queries

### CODE REVIEW

File: app/middleware/queryValidation.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { meteorsQuerySchema } from "../schemas/meteors.schema.js";
+export default function queryValidation(req, res, next) {
```
Comment (mdziarkach) [inline app/middleware/queryValidation.js]: schema should come as an argument

Comment (DiscoJordan) [reply app/middleware/queryValidation.js]: fixed, is it right now?

File: app/middleware/queryValidation.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default function queryValidation(schema) {
```
Comment (mdziarkach) [inline app/middleware/queryValidation.js]: you can provide one more argument - type ('query', 'body') here

**Review by mdziarkach — COMMENTED** [review]
looks good
you can improve it

**Pull request merged**